core/src/main/scala/schema/validator/TroySchemaValidator.scala

```diff
@@ -20,42 +20,57 @@ package schema.validator
 import cats.MonadError
 import cats.data.{NonEmptyList, Validated, ValidatedNel}
 import freestyle.cassandra.schema._
+import freestyle.cassandra.schema.provider.SchemaDefinitionProvider
 import troy.schema.{Message, Result, SchemaEngine, V}
 
-object TroySchemaValidator {
+class TroySchemaValidator[M[_]](
+    implicit M: MonadError[M, Throwable],
+    SDP: SchemaDefinitionProvider[M])
+    extends SchemaValidator[M] {
 
-  def validateStatement[M[_]](schema: SchemaDefinition, st: Statement)(
+  override def validateStatement(st: Statement)(
       implicit M: MonadError[M, Throwable]): M[ValidatedNel[SchemaError, Unit]] = {
 
-    implicit class MessageOps(message: Message) {
+    def validateStatement(
+        schema: SchemaDefinition,
+        st: Statement): M[ValidatedNel[SchemaError, Unit]] = {
 
-      def toSchemaValidatorError: SchemaValidatorError = SchemaValidatorError(message.message)
+      implicit class MessageOps(message: Message) {
```

Comment:
Not sure if the implicit classes `MessageOps` and `ResultOps` are worth it if they are only used within `validateStatement`?
A regular method doesn't have the overhead and is probably easier to understand as well.

Reply:
Totally agree